data/libs/Player.lua

```diff
@@ -281,6 +281,28 @@ function Player:ClearCrimeRecordHistory (faction)
 	self.record_old[forFaction] = nil
 end
 
+--
+-- Method: ClearCrimeRecordHistory
+--
+function Player:GetLegalStatus (faction)
+	local _, fine = self:GetCrimeRecord(faction)
+
+	local crimes, fine = Game.player:GetCrimeOutstanding()
+	Game.player:setprop("fine", fine)
+
+	if fine < 1 then
+		return('CLEAN')
+	elseif fine < 5500 then
+		return('OFFENDER')
+	elseif fine < 20000 then
+		return('CRIMINAL')
+	elseif fine < 100000 then
+		return('OUTLAW')
+	else
+		return('FUGITIVE')
+	end
+end
```

Comment:
Looking at this closer, you don't need both calls (`GetCrimeRecord` and `GetCrimeOutstanding`). As is, `fine` will be set to the value of `Game.player:GetCrimeOutstanding()` since that's the last call.

Looking at the two, you want `GetCrimeOutstanding` not `GetCrimeRecord` since the latter returns all crimes, even those that are settled.

Reply:
> Looking at this closer, you don't need both calls 

Ha, yes, it's dead code that survived my initial attempt, good catch.

> and you are already in the player class...

will do.